### PR TITLE
Trail orientatio fix

### PR DESCRIPTION
**Pull request by telmiuse**

### CODE REVIEW

File: Project/Source/Utils/Trail.h [hunk 1]
```diff
@@ -79,6 +82,7 @@ class Trail {
 
 	Quad quads[maxQuads];
 
-	Quat mainRotation = Quat(0.0f, 0.0f, 0.0f, 0.0f);
+	Quat mainRotation;
```
Comment (JaimeBea) [inline Project/Source/Utils/Trail.h:85]: The quaternions should be initialized to avoid garbage values.

**Review by JaimeBea — DISMISSED** [review]

**Review by FrancPS — DISMISSED** [review]

File: Project/Source/Utils/Trail.cpp [hunk 2]
```diff
@@ -35,12 +35,20 @@ void Trail::Init() {
...
 
+bool Trail::CalculateDistance(float3 A, float3 B) {
+	if (Abs(B.x - A.x) > vertexDistance) return true;
+	if (Abs(B.y - A.y) > vertexDistance) return true;
+	if (Abs(B.z - A.z) > vertexDistance) return true;
+	return false;
+}
 void Trail::Update(float3 mPosition) {
 	if (!isRendering) return;
+	float3 currentDirection = (previousPosition - currentPosition).Normalized();
```
Comment (lowysole) [inline Project/Source/Utils/Trail.cpp:46]: direction is  (currentPosition -previousPosition). Care then with the Perpendicular() result

File: Project/Source/Utils/Trail.cpp [hunk 3]
```diff
@@ -35,12 +35,20 @@ void Trail::Init() {
 	EditTextureCoords();
 }
 
+bool Trail::CalculateDistance(float3 A, float3 B) {
+	if (Abs(B.x - A.x) > vertexDistance) return true;
+	if (Abs(B.y - A.y) > vertexDistance) return true;
+	if (Abs(B.z - A.z) > vertexDistance) return true;
+	return false;
+}
```
[the remark this answers is not in the document]
Comment (lowysole) [reply Project/Source/Utils/Trail.cpp:43]: I checked it again and it's incorrect. The formula for 3º dimension distance is using square roots:

`if (B.Distance(A) > vertexDistance) return true;  // (B-A -> current - previous)`

or an optimal way:

`if (B.DistanceSq(A) > vertexDistance * vertexDistance) return true;`

**Review by lowysole — CHANGES_REQUESTED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by FrancPS — DISMISSED** [review]

**Comment by lowysole** [comment]
Not Compiling.

**Review by lowysole — APPROVED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Pull request merged**